TrailBlaze/Networking/Interceptor.m

```diff
@@ -0,0 +1,125 @@
+//
+//  Interceptor.m
+//  TrailBlaze
+//
+//  Created by Adam Issah on 7/20/22.
+//
+
+#import "Interceptor.h"
+#import "CoreLocation/CoreLocation.h"
+#import "MapKit/MapKit.h"
+
+@implementation Interceptor {
+    
+}
+static float etaDifferenceThreshold = 692;
+static NSMutableArray *etasDifferences;
+static NSMutableDictionary *etaPointPairs;
```

Comment:
is it necessary to define these as static instead of keeping them just scoped within `getBestETAPoint`? a problem with the current logic is that these collections are not reset after doing one getBestETAPoint calculation..

if it's necessary to keep them static you should reset them at the end of the calculation (and additionally you should amend `getBestETAPoint` so it can only have 1 in-flight calculation happening at once)

if it's not necessary i would recommend scoping these collections within the method

Reply:
> you should reset them at the end of the calculation 

also if you do these you'll want to `dispatch_async(serialQueue` your `dispatch_group_notify` block since your mutating the collections there. just for completeness of thread safety.

Reply:
I put them inside of the method, I believe I had a separate plan to break down the getBestETAPoint into smaller sub methods and that required those variables to be accessible to all functions in the class, I scrapped that idea though so they are safe inside of getBestETAPoint's method body.